src/parser.h

```diff
@@ -207,6 +207,12 @@ struct library_data_t {
 
     /// The read limit to apply to captured subshell output, or 0 for none.
     size_t read_limit{0};
+};
+
+/// Miscellaneous data used to avoid recursion and others.
+struct library_data_t {
+    /// Plain-Old-Data components that can be shared over FFI
+    struct library_data_pod pod;
```

Comment:
If we used inheritance (`library_data_t : public library_data_pod_t`), we could probably avoid the C++ changes without causing problems for Rust, right?

either way `library_data_pod` should have the `_t` suffix.